### ISSUE TITLE
A /bin/sh script executed from a /bin/{sh,bash} are do not show up in coverge report

### ISSUE THREAD

**Issue by neg** [SimonKagstrom/kcov#51]
run.sh
```bash
#!/bin/sh

./sh.sh
./bash.sh

# This shows up in coverage report
echo "foo from run"
```

sh.sh
```bash
#!/bin/sh

# This do not shows up in coverage report
echo "foo from sh"
```

bash.sh
```bash
#!/bin/bash

# This shows up in coverage report
echo "foo from bash"
```

**Comment by SimonKagstrom** [comment]
I'm guessing you're using Ubuntu or Debian? Then /bin/sh is not bash but dash.

kcov depends on bash to provide line numbers for executed code, so non-bash scripts will not work. I'll add a regression test for it, but I think it will bi difficult to solve.

**Comment by neg** [comment]
Unfortunately it's not that simple, /bin/sh is a symlink to bash.

And I do get coverage for run.sh who uses /bin/sh as its interpreter.

**Comment by SimonKagstrom** [comment]
Ok, it seems like it's caused by this problem:

http://stackoverflow.com/questions/9036057/bash-seems-to-ignore-bash-env-when-accessed-through-a-symlink

kcov uses BASH_ENV to collect coverage from child-shells, but this is ignored if bash is invoked via a /bin/sh symlink.

**Comment by neg** [comment]
You are correct and unfortunately $ENV the /bin/sh equivalent of $BASH_ENV only gets evaluated if /bin/sh is a interactive shell.

**Comment by SimonKagstrom** [comment]
I've implemented a fix for this in 5f56dc0. It uses a LD_PRELOAD:ed shared library that replaces execve, so #!/bin/sh scripts are actually executed with bash -x.

Very lightly tested yet, but closing for now.
